pkg/endpoint/policy.go

```diff
@@ -129,18 +129,45 @@ func getL4FilterEndpointSelector(filter *policy.L4Filter) []api.EndpointSelector
 	return fromEndpointsSelectors
 }
 
+func (e *Endpoint) sweepFilters(oldPolicy *policy.L4Policy,
+	identities *identityPkg.IdentityCache) (errors int) {
+
+	oldEntries, err := e.PolicyMap.DumpToSlice()
+	if err != nil {
+		e.getLogger().WithError(err).WithFields(logrus.Fields{
+			logfields.PolicyRevision: oldPolicy.Revision,
+		}).Warning("Delete stale l4 policy failed")
+		errors++
+		return
+	}
+	for _, entry := range oldEntries {
+		id := identityPkg.NumericIdentity(entry.Key.GetIdentity())
+		if _, ok := (*identities)[id]; !ok {
+			if err := e.PolicyMap.DeleteEntry(&entry); err != nil {
+				e.getLogger().WithError(err).WithFields(logrus.Fields{
+					logfields.PolicyRevision: oldPolicy.Revision,
+					logfields.Identity:       id,
+				}).Debug("Delete stale l4 policy failed")
```

Comment:
I'm concerned that if this gets in a really bad state, it will flood the logs. Hence I'm relying on the generic `"Some Label+L4 policy updates failed."`.

Reply:
Should we ever see this is the previous commit is working and the system is behaving normal?

Reply:
No, I would expect not.

Reply:
I'm OK to do a warning then, I'd rather be sure that we see this and the user yells.

Reply:
OK, I'll update this after I see the CI results.